src/hooks/screen/useCreateChat.ts

```diff
@@ -33,33 +50,62 @@ const useCreateChat = () => {
       console.log(e, 'error on memberSchema');
     }
   };
-  const createChannelJson = (response, selectedUser) => {
+  const createChannelJson = (response, selectedUser, isAnonymous = false) => {
     const channelWithMember = {...response.channel, members: response.members};
+    let targetName =
+      selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username;
+    let targetImage =
+      selectedUser?.user?.profilePicture ||
+      selectedUser?.user?.image ||
+      selectedUser.profilePicture;
+
     const targetRawJson = {
       type: 'notification.message_new',
       cid: response?.channel?.id,
       channel_id: '',
       channel_type: 'messaging',
       channel: channelWithMember,
       created_at: response?.channel,
-      targetName: selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username,
-      targetImage:
-        selectedUser?.user?.profilePicture ||
-        selectedUser?.user?.image ||
-        selectedUser.profilePicture
+      targetName,
+      targetImage
+    };
+
+    let defaultAnonUserInfo: AnonUserInfoTs = {
+      anon_user_info_color_code: '',
+      anon_user_info_color_name: '',
+      anon_user_info_emoji_code: '',
+      anon_user_info_emoji_name: ''
     };
-    const chatData = {
+
+    if (isAnonymous) {
+      const {
+        anonUserInfoColorCode,
+        anonUserInfoColorName,
+        anonUserInfoEmojiCode,
+        anonUserInfoEmojiName,
+        channelImage,
+        channelName
+      } = getChannelListInfo(response?.channel, signedProfileId, anonProfileId);
+      defaultAnonUserInfo = {
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      };
+
+      targetName = channelName;
+      targetImage = channelImage;
+    }
+
+    const chatData: MessageAnonymouslyData = {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code seems to be handling user data and creating a JSON object based on it. However, there's no null/undefined check for `response` before accessing its properties. This could potentially lead to a TypeError if `response` is undefined. Please add necessary null checks.
```diff
- const channelWithMember = {...response.channel, members: response.members};
+ const channelWithMember = response ? {...response.channel, members: response.members} : {};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->